webapp-django/crashstats/signature/views.py

```diff
@@ -35,36 +37,36 @@
 )
 
 
-def get_validated_params(request):
-    params = get_params(request)
-    if isinstance(params, http.HttpResponseBadRequest):
-        # There was an error in the form, let's return it.
-        return params
+def get_validated_params(view):
+    @functools.wraps(view)
+    def inner(request, *args, **kwargs):
+        try:
+            params = get_params(request)
 
-    if len(params['signature']) > 1:
-        return http.HttpResponseBadRequest(
-            'Invalid value for "signature" parameter, '
-            'only one value is accepted'
-        )
+            if len(params['signature']) > 1:
+                raise ValidationError(
+                    'Invalid value for "signature" parameter, '
+                    'only one value is accepted'
+                )
 
-    if not params['signature'] or not params['signature'][0]:
-        return http.HttpResponseBadRequest(
-            '"signature" parameter is mandatory'
-        )
+            if not params['signature'] or not params['signature'][0]:
+                raise ValidationError(
+                    '"signature" parameter is mandatory'
+                )
+        except ValidationError, e:
+            return http.HttpResponseBadRequest(str(e))
 
-    return params
+        kwargs['params'] = params
+        return view(request, *args, **kwargs)
+    return inner
 
 
 @waffle_switch('signature-report')
+@get_validated_params
 @pass_default_context
```

Comment:
I wonder if this is needed? Isn't that just there so that you can get access to the current products and versions. Try removing it (and the `default_context=None`) and see if it all continues to work.